### PR TITLE
HTML decoding misses

### PR DESCRIPTION
**Pull request by yb66**
Hi,

I didn't finish writing the pull request last time before you responded. I've done this because it was missing HTML entities in some code that I've been using, which I noticed as I was porting it over from Nokogiri to Oga.

Hope it helps.

iain

### CODE REVIEW

**Comment by yb66** [comment]
I've changed the regex. There was no need to add a new failure spec as one of the ones I'd already done catches it, which is nice ;-)

File: lib/oga/entity_decoder.rb [hunk 1]
```diff
@@ -13,6 +13,7 @@ def self.try_decode(input, html = false)
     # @return [String]
     #
     def self.decode(input, html = false)
+      fail ArgumentError, "decode was sent nil" if input.nil?
```
Comment (YorickPeterse) [inline lib/oga/entity_decoder.rb]: Why is this added? The code in `Oga::EntityDecoder.try_decode` already makes sure this particular `decode` method never receives `nil` as input.

Comment (YorickPeterse) [inline lib/oga/entity_decoder.rb]: Also, if this were needed it would still need some changes:

* Use `raise` instead of `fail`. I'm aware tools such as Rubocop advocate `fail` for byzantine reasons, but Oga uses `raise` everywhere and I like to keep it that way.
* The error message itself is not particularly useful.
* Changes such as these _require_ specs to be added to cover the new behaviour, these are currently missing.

**Comment by YorickPeterse** [comment]
As another note, Oga uses the RSpec 2 "should" syntax instead of the "expect" syntax. I really, _really_ hate having to write `expect(x).to eq(y)` instead of just `x.should == y`. I've never run into a problem with RSpec's `should` colliding with some other library, until that time I want to continue using the "should" syntax.

**Comment by yb66** [comment]
I prefer `should` too, but I wasn't going to arse around with using non-standard syntax like `lambda` to write a one line spec for failure. YMMV

File: lib/oga/entity_decoder.rb (hunk 1 above)
Comment (yb66) [inline lib/oga/entity_decoder.rb]: It's `fail` because it's a failure, not a re-raise. I'm not raising the error, it's failing a contract.

`try_decode` does check for this (though I couldn't see a reason why it's structured this way, a comment would be nice) but that's the point - if it gets this fair it should fail. Hence why I didn't add anything to `try_decode`, even though it should probably have more argument checks.

**Comment by YorickPeterse** [comment]
I'm closing this pull request. Judging by your comments and disregard for my requests it seems you're not interest in putting any real effort into this. As such I have no real interest in merging this and then having to fix everything, I'll just fix this problem myself.
